Lib/test/test_statistics.py

```diff
@@ -1004,6 +1004,9 @@ def test_nan(self):
             x = statistics._convert(nan, type(nan))
             self.assertTrue(_nan_equal(x, nan))
 
+    def test_raise_type_error(self):
+        self.assertRaises(TypeError, statistics._convert, None, float)
```

Comment:
I find using `with self.assertRaises():` much more readable, please consider changing this to that style.

```suggestion
        with self.assertRaises(TypeError):
            statistics._convert(None, float)
```

(Also in following new tests in this PR.)

Reply:
Sure, I will change it here, as well as in the `FindLteqTest` and `FindRteqTest` tests, as you suggested.

Reply:
Sorry to mention it just after you made the change already, but I find that using `assertRaisesRegex()` tends to be a better practice than just `assertRaises()`. It's more significant of an issue for things like deprecation warnings, but it can help quite a bit with ensuring that you're getting the correct exception.

So, it may not be necessary in the above case, but it's worth some consideration.

Reply:
> I find that using assertRaisesRegex() tends to be a better practice than just assertRaises().

That may be true in general, but I think that in this case it's most appropriate to simply check that a TypeError is thrown.

Reply:
Yeah, looking at the above test a bit more closely, it seems highly unlikely for any other `TypeError`s to occur besides the specific one being tested. So, `assertRaises()` is probably fine here.

Reply:
Thanks for your suggestion @aeros. As you have both already said, it may not be useful in this particular test, but maybe, it would be useful in the `test_single_value_unsupported_type` test (introduced in this PR) because it targets a specific `TypeError` with a custom message, while it is possible for the function to raise other TypeErrors too (through the `_fail_neg` function).

If you don't have any objections, I will go and modify the `test_single_value_unsupported_type` to use the `assertRaisesRegex` instead of the `assertRaises`.

Reply:
> f you don't have any objections, I will go and modify the `test_single_value_unsupported_type` to use the `assertRaisesRegex` instead of the `assertRaises`.

Go ahead.